text/0000-action-menu.md

```diff
@@ -0,0 +1,214 @@
+
+## Purpose
+
+1. **Improve Usability:** It provides a direct and straightforward way for users to navigate the application and perform tasks. It simplifies the user interface, improving the overall usability of the application.
+
+2. **Consolidate Actions:** It consolidates various options or actions into a single, organized element, making it easier for users to find the functionality they need without having to navigate through different parts of the application.
+
+3. **Streamline User Interactions:** An ActionMenu provides an organized structure where all related actions are grouped. This streamlining enhances the user's efficiency as they can quickly find the necessary action within a logically sorted list, resulting in a more intuitive user experience.
+
+## Benefits
+- **Modularity and Customizability:** The proposed headless ActionMenu promotes modular design and allows for extensive customization, fitting seamlessly into any design aesthetics.
+
+- **Accessibility Enhancement:** The commitment to improve accessibility ensures that all Pando users, regardless of their abilities, can effortlessly interact with the platform, thereby promoting inclusivity.
+
+- **Multi-Framework Support:** The headless ActionMenu's development will support various JavaScript frameworks, including React, Svelte, Angular, and Vue. This cross-framework compatibility ensures that developers working with any of these frameworks can seamlessly incorporate the ActionMenu into their projects.
+
+- **Streamlining Development:** This feature introduces a standardized way to create and manage ActionMenus, reducing the time and resources needed for individual development tasks.
+
+# Basic example
+
+Here is a basic example using TailwindCSS for styles:
+
+```jsx
+<ActionMenu className="relative">
+    <ActionMenu.Button className="text-black bg-white" aria-label="Open menu">
+      Open menu
+    </ActionMenu.Button>
+    <ActionMenu.Overlay>
+      <ActionList className="absolute bg-gray-700">
+        <ActionList.Item onSelect={() => alert("Copy link clicked")}>
+          Copy link
+          <ActionList.TrailingVisual>⌘C</ActionList.TrailingVisual>
+        </ActionList.Item>
+        <!-- more items... -->
+      </ActionList>
+    </ActionMenu.Overlay>
+</ActionMenu>
+```
+## Edge Cases
+Here are a few potential corner-cases for this ActionMenu component:
+
+1. **Empty ActionMenu**: What happens when an ActionMenu is rendered without any ActionList.Items? Your component should handle this case gracefully, possibly rendering an empty state message or not rendering the ActionMenu at all.
+
+2. **A very long ActionList.Item**: What if one of your ActionList.Item's has a very long string or a complex node structure? The ActionMenu should handle this without breaking the layout or causing horizontal scrolling.
```

Comment:
I actually think this is probably not the responsibility of the `ActionMenuList` because it creates a restriction which you want the user to decide when that is and not the library.

Reply:
This makes sense. I've updated the 'Edge Cases' section accordingly.